source/server/server.cc

```diff
@@ -151,24 +151,36 @@ void InstanceImpl::getParentStats(HotRestart::GetParentStatsInfo& info) {
 bool InstanceImpl::healthCheckFailed() { return server_stats_->live_.value() == 0; }
 
 void InstanceUtil::loadBootstrapConfig(envoy::config::bootstrap::v2::Bootstrap& bootstrap,
-                                       const std::string& config_path, bool v2_only) {
-  bool v2_config_loaded = false;
+                                       Options& options) {
   try {
-    MessageUtil::loadFromFileAndValidate(config_path, bootstrap);
-    v2_config_loaded = true;
+    if (!options.configPath().empty()) {
+      MessageUtil::loadFromFileAndValidate(options.configPath(), bootstrap);
+    }
+    if (!options.configYaml().empty()) {
+      auto merge = envoy::config::bootstrap::v2::Bootstrap();
+      MessageUtil::loadFromYamlAndValidate(options.configYaml(), merge);
+      bootstrap.MergeFrom(merge);
+    }
+    return;
   } catch (const EnvoyException& e) {
-    if (v2_only) {
+    if (options.v2ConfigOnly()) {
       throw;
-    } else {
-      // TODO(htuch): When v1 is deprecated, make this a warning encouraging config upgrade.
-      ENVOY_LOG(debug, "Unable to initialize config as v2, will retry as v1: {}", e.what());
     }
+    // TODO(htuch): When v1 is deprecated, make this a warning encouraging config upgrade.
+    ENVOY_LOG(debug, "Unable to initialize config as v2, will retry as v1: {}", e.what());
   }
-  if (!v2_config_loaded) {
-    Json::ObjectSharedPtr config_json = Json::Factory::loadFromFile(config_path);
+  Json::ObjectSharedPtr config_json;
+  if (!options.configPath().empty()) {
+    config_json = Json::Factory::loadFromFile(options.configPath());
     Config::BootstrapJson::translateBootstrap(*config_json, bootstrap);
-    MessageUtil::validate(bootstrap);
   }
+  if (!options.configYaml().empty()) {
+    config_json = Json::Factory::loadFromYamlString(options.configYaml());
+    auto merge = envoy::config::bootstrap::v2::Bootstrap();
+    Config::BootstrapJson::translateBootstrap(*config_json, merge);
+    bootstrap.MergeFrom(merge);
```

Comment:
I think we can skip doing this for v1, since v1 is feature frozen effectively unless there is a strong case for a new feature.

Reply:
Done.